added rgb<->HER color conversion (#7397)

eokoshi opened the pull request:
## Description

Addresses #7397

- Previous `rgb2hed` conversion between RGB and and HED space was more suited for stains with DAB staining included.
- This commit adds the default stain vectors from QuPath for hematoxylin, eosin, and residual.
- added `rgb2her` and `her2rgb` functions to `colorconv.py` (only slight modifications from rgb2hed/hed2rgb functions)
- added disclaimer from QuPath about color deconvolution
- added to `__init__.pyi` to make new additions importable from `skimage.color`
- added usage example to gallery (slight modifications to `plot_ihc_color_separation.ipynb`)
- added tests (copied from `rgb2hed`)

eokoshi committed:
added rgb<->HER color conversion (#7397)

Addresses #7397

- Previous `rgb2hed` conversion between RGB and and HED space was more
suited for stains with DAB staining included.
- This commit adds the default stain vectors from QuPath for
hematoxylin, eosin, and residual.
- added `rgb2her` and `her2rgb` functions to `colorconv.py` (only slight
modifications from rgb2hed/hed2rgb functions)
- added disclaimer from QuPath about color deconvolution
- added to `__init__.pyi` to make new additions importable from
`skimage.color`
- added usage example to gallery (slight modifications to
`plot_ihc_color_separation.ipynb`)
- added tests (copied from `rgb2hed`)

eokoshi committed:
Merge branch 'main' into colorconv-HER

eokoshi committed:
Updated name from HER space to HEX space

- Also fixed previous error where rgb2hex/hex2rgb functions used hed
matrix instead of correct hex matrix.
- implemented residual channel calculation as np.cross rather than hard
coding it.

eokoshi committed:
updated HER to HEX naming in tests

mkcor commented on doc/examples/color_exposure/plot_rgb_to_hex.py:
```diff
@@ -0,0 +1,98 @@
+fig.tight_layout()
+
+######################################################################
+# Now we can easily manipulate the hematoxylin and eosin channels:
```
```suggestion
# Now we can manipulate the hematoxylin and eosin channels conveniently:
```

matthew-brett commented:
We have changed the base branch for this PR from the `main` branch to the `pre-megamove-integration` branch.

## Summary - what this means for your PR

You can continue to work on your PR, now based on `pre-megamove-integration`.  You should update your branch to the current state of `pre-megamove-integration` (this is what the Github "Update branch" button does).   Please **do not merge the scikit-image `main` branch into your PR branch**.

Once we've merged your PR into the `pre-megamove-integration` branch, we'll update your changes for Megamove, and merge into the `main` branch.

Read on for the rationale, and more detailed instructions.

<details>

## Rationale

We've changed the base branch to make it easier and smoother to continue to work on your PR, while we work through a rather large change in the directory structure of the repository - a change that we have called *Megamove*.

See [the Megamove explanation issue](https://github.com/scikit-image/scikit-image/issues/8232) for more detail.

Changing the base branch means you do not see merge conflicts related to the *Megamove* change, and can continue to work with the repository in its original state, without worrying about porting those changes across to the new post-Megamove structure.

As you'll see in the linked issue, you can continue to work on your PR as before, and you can continue to use the Github GUI to "Update branch" - this updates from the new `pre-megamove-integration` branch.   However, there is one caveat: please, do not try and merge the `scikit-image` `main` branch into this one, but instead, keep track of the `pre-megamove-integration` branch.  The issue explains more.

To emphasize again, **do not try and merge the scikit-image `main` branch**.  You will get Megamove related merge conflicts that are rather tricky to resolve correctly.

When we've approved your PR, we'll merge into the `pre-megamove-integration` branch, and then process it so we can merge it into the `main` branch.   We'll let you know when we've merged your PR changes into `main`.

</details>

Any questions, please reply here.

eokoshi committed:
Merge branch 'pre-megamove-integration' into colorconv-HER

eokoshi committed:
Update src/skimage/color/colorconv.py

link formatting in docs

Co-authored-by: Marianne Corvellec <marianne.corvellec@ens-lyon.org>

eokoshi committed:
change wording doc/examples/color_exposure/plot_rgb_to_hex.py

Co-authored-by: Marianne Corvellec <marianne.corvellec@ens-lyon.org>

eokoshi commented on src/skimage/color/colorconv.py:
```diff
@@ -624,8 +618,7 @@ def xyz_tristimulus_values(*, illuminant, observer, dtype=float):
         raise ValueError(
-            f'Unknown illuminant/observer combination '
-            f'(`{illuminant}`, `{observer}`)'
+            f'Unknown illuminant/observer combination (`{illuminant}`, `{observer}`)'
```
yes, they were

eokoshi committed:
import ski instead of modules separately

eokoshi committed:
fix spelling error

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

eokoshi commented:
I implemented the suggested changes from @mkcor thank you.

I would like to mention that the doc/examples file is copied directly from `doc/examples/color_exposure/plot_ihc_color_separation.py" with appropriate modifications. 

Could someone let me know what is necessary to get this merged?
Thank you

eokoshi committed:
improve references

Pinned QuPath code links to v0.7.0 from main
added suggested reference from @petebankhead

eokoshi commented:
I used HEX on recommendation to align with another __X object naming convention.

Should I change this back to HER?
